packages/core/src/components/TextField/Mask.jsx

```diff
@@ -75,6 +75,7 @@ function toDigitsAndAsterisks(value) {
  */
 function toNumber(value) {
   if (typeof value !== 'string') return value;
+  if (!value.match(/\d/)) return undefined;
```

Comment:
This can still cause errors when entering things like `11.` or `11. ...` or `11.bbb`, etc. Because we split on `.` later on, it causes an error when periods are included along with invalid info later, because the `parts[1].match(digitsRegex)` becomes `null` in that case and `join` can't be called on it.

Here is one solution (I think...) that works for that edge case:
```
  let b = '';
  if (parts.length >= 2 && parts[1].match(digitsRegex)) {
    b = parts[1].match(digitsRegex).join('');
  }
```

Reply:
Ugh, this whole thing seems more complicated than it needs to be. Maybe I should rethink all these masking functions. That ugh isn't at you, and I think your fix will help. Thanks for reviewing this